extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/RestClientBuilderHelper.java

```diff
@@ -112,4 +110,58 @@ public static Sniffer createSniffer(RestClient client, ElasticsearchConfig confi
 
         return builder.build();
     }
+
+    public enum EsAuth {
+        NONE {
+            @Override
+            public void apply(HttpAsyncClientBuilder httpClientBuilder, ElasticsearchConfig config) {
+                // No authentication needed
+            }
+        },
+
+        BASIC {
+            @Override
+            public void apply(HttpAsyncClientBuilder httpClientBuilder, ElasticsearchConfig config) {
+                if (config.username().isPresent()) {
+                    if (!"https".equalsIgnoreCase(config.protocol())) {
+                        LOG.warn("Using Basic authentication in HTTP implies sending plain text passwords over the wire, "
+                                + "use the HTTPS protocol instead.");
+                    }
+
+                    CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+                    credentialsProvider.setCredentials(AuthScope.ANY,
+                            new UsernamePasswordCredentials(config.username().get(),
+                                    config.password().orElse(null)));
+                    httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+                }
+            }
+        },
+
+        API_KEY {
+            @Override
+            public void apply(HttpAsyncClientBuilder httpClientBuilder, ElasticsearchConfig config) {
+                if (config.esApiKeyAuth().isPresent()) {
+                    ElasticsearchConfig.EsApiKeyAuth auth = config.esApiKeyAuth().get();
+                    Header apiKeyHeader = new BasicHeader(HttpHeaders.AUTHORIZATION,
+                            "ApiKey " + auth.apiKey());
+                    httpClientBuilder.setDefaultHeaders(Collections.singleton(apiKeyHeader));
+                    LOG.info("API Key authentication is enabled.");
+                }
+            }
+        };
+
+        public abstract void apply(HttpAsyncClientBuilder httpClientBuilder, ElasticsearchConfig config);
+    }
+
+    private static EsAuth checkAuthMethod(ElasticsearchConfig config) {
+        boolean hasBasic = config.username().isPresent();
+        boolean hasApiKey = config.esApiKeyAuth().isPresent();
+
+        if (hasBasic && hasApiKey) {
+            LOG.warn("Multiple authentication methods configured. Defaulting to Basic Authentication.");
+            return EsAuth.BASIC;
+        }
```

Comment:
I'd still say it's better to throw an exception here instead of logging a warning since this is a "misconfiguration", and a user would want to be precise about the auth methods.